integration_test/third_party_apps_data/applications/dcgm/centos_rhel/install

```diff
@@ -1,54 +1,84 @@
+verify_driver() {
+    # check NVIDIA driver installation succeeded
+    nvidia-smi
+}
 
-# check NVIDIA driver installation succeeded
-nvidia-smi
+install_driver_from_runfile() {
+    # Ref: https://docs.nvidia.com/datacenter/tesla/tesla-installation-notes/index.html#runfile
+    # This method requires the matching kernel-devel package to be installed, and 
+    # the package may be absent from the repo and cause this method to fail
+    sudo yum install -y kernel-devel-$(uname -r) pciutils gcc make wget yum-utils 
+    local DRIVER_VERSION=535.129.03
+    echo "Installing NVIDIA Data Center driver $DRIVER_VERSION"
+    curl -fSsl -O https://us.download.nvidia.com/tesla/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
+    sudo bash ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --silent
+    verify_driver
+}
 
-# Install DCGM
-VERSION_ID=${VERSION_ID%%.*}
-sudo yum-config-manager \
-    --add-repo http://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
-sudo yum clean expire-cache
-sudo yum install -y datacenter-gpu-manager
-sudo systemctl --now enable nvidia-dcgm
-
-# check DCGM service running and load profiling module
-dcgmi discovery --list
+setup_repo() {
+    sudo yum install -y yum-utils
+    sudo yum-config-manager \
+        --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+    sudo yum clean all
+}
+
+install_dcgm() {
+    # Ref: https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#rhel-centos-rocky-linux
+    setup_repo
+    sudo yum install -y datacenter-gpu-manager
+    sudo systemctl --now enable nvidia-dcgm 
+
+    # check DCGM service running and load profiling module
+    dcgmi discovery --list
+}
+
+try_install() {
+    for install_method in "$@"; do
+        echo "Installing NVIDIA driver with $install_method..."
+        # Temporarily disable the "-e" option so that failures in the function
+        # won't cause this script to exit; also run the installation method in a
+        # subshell with -e so that the function itself will exit immediately if 
+        # any command encounters error
+        set +e ; ( set -e; $install_method ) ; exit_code=$?
+        set -e
+        if [ $exit_code -eq 0 ]; then
+            echo "NVIDIA driver has been installed successfully with $install_method."
+            return 
+        fi
+    done
+    echo "NVIDIA driver cannot be installed; all installation methods failed."
+    return 1
+}
+
+
+handle_rhel7() {
+    install_driver_from_package_manager() {
+        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=7&target_type=rpm_network
+        setup_repo
+        sudo yum -y install nvidia-driver-latest-dkms
+        verify_driver
+    }
+
+    try_install install_driver_from_runfile install_driver_from_package_manager
```

Comment:
If there's some documentation that indicates that installing from a runfile is preferable to installing the system packages on rhel7, this would be a good place to cite it (e.g.:
```
    # Prefer installing from a runfile on RHEL 7: http://…
```
).

Reply:
Looked into this a bit more and it seems that unlike Rocky 8/9, Centos 7's repo would have all versions kernel source packages (https://pkgs.org/download/kernel-devel); installing GPU driver from yum would function perfectly on Centos 7 if we install the kernel source first. 

Updated the script so that we install from package managers for all three distros, and leave runfile as the backup method.